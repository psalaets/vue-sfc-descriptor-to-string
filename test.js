const fs = require('fs');
const compiler = require('vue-template-compiler');

const toString = require('./index');

it('vanilla sfc', async () => {
  const descriptor = await parse('./test-components/Vanilla.vue');

  const result = toString(descriptor);

  expect(result).toMatchSnapshot();
});

function parse(pathToComponent) {
  return new Promise((resolve, reject) => {
    fs.readFile(pathToComponent, 'utf8', (err, sfc) => {
      if (err) return reject(err);

      try {
        const descriptor = compiler.parseComponent(sfc);
        return resolve(descriptor);
      } catch (e) {
        return reject(e);
      }
    })
  });
}