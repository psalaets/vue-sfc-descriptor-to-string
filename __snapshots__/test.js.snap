// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vanilla sfc 1`] = `
"<script>
export default {
  props: {
    name: {
      type: String
    }
  }
};
</script>

<template>
  <div class=\\"widget\\">Hello {{name}}</div>
</template>

<style>
.widget {
  color: red;
}
</style>
"
`;
